File: src/ru/galakart/majordroid/ControsActivity.java
package ru.galakart.majordroid;

import android.app.Activity;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.text.method.ScrollingMovementMethod;
import android.view.Gravity;
import android.view.Menu;
import android.view.View;
import android.view.WindowManager;
import android.webkit.HttpAuthHandler;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.TextView;
import android.widget.Toast;

public class ControsActivity extends Activity {
	private WebView webPost;
	private String serverURL, login, passw, pathScripts = "";
	private boolean outAccess = false;
	private SharedPreferences prefs;
	private TextView textView_legend;
	private String scriptnames[] = new String[9];

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		SharedPreferences prefs = PreferenceManager
				.getDefaultSharedPreferences(this);
		String vid = prefs.getString(getString(R.string.vid), "");
		pathScripts = prefs.getString(getString(R.string.path_scripts), "");
		if (vid.contains("Обычный")) {
			getWindow().addFlags(
					WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
			getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
		}
		if (vid.contains("Полноэкранный")) {
			getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
			getWindow().clearFlags(
					WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
		}
		setContentView(R.layout.activity_contros);
		webPost = (WebView) findViewById(R.id.webPost);
		webPost.getSettings().setJavaScriptEnabled(true);
		webPost.setWebViewClient(new MajorDroidWebViewer());
		textView_legend = (TextView) findViewById(R.id.textView_legend);
		textView_legend.setMovementMethod(new ScrollingMovementMethod());
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.contros, menu);
		return true;
	}

	private class MajorDroidWebViewer extends WebViewClient {
		@Override
		public boolean shouldOverrideUrlLoading(WebView view, String url) {
			view.loadUrl(url);
			return true;
		}

		@Override
		public void onReceivedHttpAuthRequest(WebView view,
				HttpAuthHandler handler, String host, String realm) {
			if (outAccess)
				handler.proceed(login, passw);
		}
	}

	public void onResume() {
		super.onResume();
		prefs = PreferenceManager.getDefaultSharedPreferences(this);

		String localURL = prefs.getString(getString(R.string.localUrl), "");
		String globalURL = prefs.getString(getString(R.string.globalUrl), "");
		String dostup = prefs.getString(getString(R.string.dostup), "");
		login = prefs.getString(getString(R.string.login), "");
		passw = prefs.getString(getString(R.string.passw), "");
		scriptnames[0] = prefs.getString(getString(R.string.scriptname1), "");
		scriptnames[1] = prefs.getString(getString(R.string.scriptname2), "");
		scriptnames[2] = prefs.getString(getString(R.string.scriptname3), "");
		scriptnames[3] = prefs.getString(getString(R.string.scriptname4), "");
		scriptnames[4] = prefs.getString(getString(R.string.scriptname5), "");
		scriptnames[5] = prefs.getString(getString(R.string.scriptname6), "");
		scriptnames[6] = prefs.getString(getString(R.string.scriptname7), "");
		scriptnames[7] = prefs.getString(getString(R.string.scriptname8), "");
		scriptnames[8] = prefs.getString(getString(R.string.scriptname9), "");
		if (dostup.contains("Локальный")) {
			outAccess = false;
			serverURL = localURL;
		} else if (dostup.contains("Глобальный")) {
			outAccess = true;
			serverURL = globalURL;
		}
		String legend = "";
		for (int i = 0; i < 9; i++) {
			if (scriptnames[i] != "")
				legend += i + 1 + ". " + scriptnames[i] + ".\n";
			else
				legend += i + 1 + ". Скрипт не задан.\n";
		}
		textView_legend.setText(legend);
	}

	public void imgb_script1_exec(View v) {
		String script = scriptnames[0];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script2_exec(View v) {
		String script = scriptnames[1];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script3_exec(View v) {
		String script = scriptnames[2];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script4_exec(View v) {
		String script = scriptnames[3];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script5_exec(View v) {
		String script = scriptnames[4];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script6_exec(View v) {
		String script = scriptnames[5];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script7_exec(View v) {
		String script = scriptnames[6];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script8_exec(View v) {
		String script = scriptnames[7];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void imgb_script9_exec(View v) {
		String script = scriptnames[8];
		if (script != "")
			scriptExec(script);
		else
			scriptneni();
	}

	public void scriptExec(String script) {
		webPost.loadUrl("http://" + serverURL + pathScripts + script);
		Toast toast = Toast.makeText(getApplicationContext(), "Скрипт "
				+ script, Toast.LENGTH_LONG);
		toast.setGravity(Gravity.BOTTOM, 0, 0);
		toast.show();
	}

	private void scriptneni() {
		Toast toast = Toast
				.makeText(
						getApplicationContext(),
						"Кнопка не привязана к скрипту.\nЗадайте имя скрипта в настройках",
						Toast.LENGTH_LONG);
		toast.setGravity(Gravity.BOTTOM, 0, 0);
		toast.show();
	}
}

File: app/src/main/java/ru/galakart/majordroid/ControsActivity.kt
package ru.galakart.majordroid

import android.app.Activity
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.text.method.ScrollingMovementMethod
import android.view.Gravity
import android.view.View
import android.view.WindowManager
import android.webkit.HttpAuthHandler
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_contros.*

class ControsActivity : Activity() {
    private var serverURL = ""
    private var login = ""
    private var passw = ""
    private var pathScripts = ""
    private var outAccess = false
    private val scriptnames = arrayOfNulls<String>(9)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_contros)
        setScreenAndComponents()
    }

    private inner class MajorDroidWebViewer : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
            view!!.loadUrl(request!!.url.toString())
            return true
        }
        override fun onReceivedHttpAuthRequest(view: WebView, handler: HttpAuthHandler, host: String, realm: String) {
            if (outAccess)
                handler.proceed(login, passw)
        }
    }

    public override fun onResume() {
        super.onResume()
        loadSettings()
    }

    private fun setScreenAndComponents() {
        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        val vid = prefs.getString("view", "")
        when (vid) {
            "1" -> {
                window.addFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN)
                window.clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
            }
            "2" -> {
                window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN)
                window.clearFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN)
            }
        }
        webPost!!.webViewClient = MajorDroidWebViewer()
        tvLegend!!.movementMethod = ScrollingMovementMethod()
    }

    private fun loadSettings() {
        var prefs = PreferenceManager.getDefaultSharedPreferences(this)
        val localURL = prefs!!.getString("localaddress", "")
        val globalURL = prefs!!.getString("globaladdress", "")
        val dostup = prefs!!.getString("access", "")
        login = prefs!!.getString("login", "")
        passw = prefs!!.getString("passw", "")
        pathScripts = prefs!!.getString("scriptprocessor", "")
        scriptnames[0] = prefs!!.getString("scriptname1", "")
        scriptnames[1] = prefs!!.getString("scriptname2", "")
        scriptnames[2] = prefs!!.getString("scriptname3", "")
        scriptnames[3] = prefs!!.getString("scriptname4", "")
        scriptnames[4] = prefs!!.getString("scriptname5", "")
        scriptnames[5] = prefs!!.getString("scriptname6", "")
        scriptnames[6] = prefs!!.getString("scriptname7", "")
        scriptnames[7] = prefs!!.getString("scriptname8", "")
        scriptnames[8] = prefs!!.getString("scriptname9", "")
        when (dostup) {
            "1" -> {
                outAccess = false
                serverURL = localURL
            }
            "2" -> {
                outAccess = true
                serverURL = globalURL
            }
        }
        var legend = ""
        for (i in 0..8) {
            legend += if (scriptnames[i] !== "")
                (i + 1).toString() + ". " + scriptnames[i] + ".\n"
            else
                (i + 1).toString() + ". " + getString(R.string.noscript) + ".\n"
        }
        tvLegend!!.text = legend
    }

    fun View.imgbScript1Exec() {
        scriptExec(scriptnames[0])
    }

    fun View.imgbScript2Exec() {
        scriptExec(scriptnames[1])
    }

    fun View.imgbScript3Exec() {
        scriptExec(scriptnames[2])
    }

    fun View.imgbScript4Exec() {
        scriptExec(scriptnames[3])
    }

    fun View.imgbScript5Exec() {
        scriptExec(scriptnames[4])
    }

    fun View.imgbScript6Exec() {
        scriptExec(scriptnames[5])
    }

    fun View.imgbScript7Exec() {
        scriptExec(scriptnames[6])
    }

    fun View.imgbScript8Exec() {
        scriptExec(scriptnames[7])
    }

    fun View.imgbScript9Exec() {
        scriptExec(scriptnames[8])
    }

    private fun scriptExec(script: String?) {
        if (!script.isNullOrEmpty()) {
            webPost!!.loadUrl("http://$serverURL$pathScripts$script")
            val toast = Toast.makeText(
                            applicationContext,
                       getString(R.string.scriptok) + "($script)",
                            Toast.LENGTH_LONG)
            toast.setGravity(Gravity.BOTTOM, 0, 0)
            toast.show()
        } else {
            val toast = Toast.makeText(
                            applicationContext,
                            getString(R.string.scriptnotset),
                            Toast.LENGTH_LONG)
            toast.setGravity(Gravity.BOTTOM, 0, 0)
            toast.show()
        }
    }
}
